Documentation format revisions. (#1505)
vircadia/vircadia: BUILD_LINUX.md

You can use the [Vircadia Builder](https://github.com/vircadia/vircadia-builder) to build on Linux more easily. Alternatively, you can follow the manual steps below.

## Ubuntu 18.04

### Ubuntu 18.04 Server only
Add the universe repository:  
_(This is not enabled by default on the server edition.)_
```bash
sudo add-apt-repository universe
sudo apt-get update
```

### Install build tools:
-  First update the repositories:  
```bash
sudo apt-get update -y
sudo apt-get upgrade -y
```

-  Install git
```bash
sudo apt-get install git -y
```
Verify git was installed by running `git --version`.

-  Install g++
```bash
sudo apt-get install g++ -y
```
Verify g++ was installed by running `g++ --version`.

-  **Ubuntu 18.04** CMake
```bash
sudo apt-get install cmake -y
```
Verify CMake was installed by running `cmake --version`.

### Install build dependencies:
-  OpenSSL:
```bash
sudo apt-get install libssl-dev
```
Verify OpenSSL was installed by running `openssl version`.

- OpenGL:
```bash
sudo apt-get install libgl1-mesa-dev -y
```
Verify OpenGL:
  - First install mesa-utils with the command `sudo apt install mesa-utils -y`.
  - Then run `glxinfo | grep "OpenGL version"`.

> ### Extra dependencies to compile an Interface on a server

Reviewer: ```suggestion
### Extra dependencies to compile Interface on a server
```

# Build Linux

*Last Updated on December 1, 2021*

Please read the [general build guide](BUILD.md) for information on dependencies required for all platforms. Only Linux specific instructions are found in this file.

You can use the [Vircadia Builder](https://github.com/vircadia/vircadia-builder) to build on Linux more easily. Alternatively, you can follow the manual steps below.

## Ubuntu 16.04/18.04

### Ubuntu 16.04 only

Add the following line to `.bash_profile`:

```bash
export QT_QPA_FONTDIR=/usr/share/fonts/truetype/dejavu/

> ```

Reviewer 2: Since we don't support Ubuntu 16.04 and Ubuntu 16.04 is EOL *and* the instructions are incomplete *and* they don't tell us what they are trying to accomplish, this part should be removed.

### Ubuntu 18.04 Server only
Add the universe repository:  
_(This is not enabled by default on the server edition.)_
```bash
sudo add-apt-repository universe
sudo apt-get update
```

#### Install build tools:
-  First update the repositories:  
```bash
sudo apt-get update -y
sudo apt-get upgrade -y
```

-  Install git
```bash
sudo apt-get install git -y
```
Verify git was installed by running `git --version`.

-  Install g++
```bash
sudo apt-get install g++ -y
```
Verify g++ was installed by running `g++ --version`.

-  **Ubuntu 18.04** CMake
```bash
sudo apt-get install cmake -y
```
Verify CMake was installed by running `cmake --version`.

- **Ubuntu 16.04** CMake  
```bash
wget https://cmake.org/files/v3.14/cmake-3.14.2-Linux-x86_64.sh
sudo sh cmake-3.14.2-Linux-x86_64.sh --prefix=/usr/local --exclude-subdir
```

#### Install build dependencies:
-  OpenSSL:
```bash
sudo apt-get install libssl-dev
```
Verify OpenSSL was installed by running `openssl version`.

- OpenGL:
```bash
sudo apt-get install libgl1-mesa-dev -y

> sudo ln -s /usr/lib/x86_64-linux-gnu/libGL.so.346.35 /usr/lib/x86_64-linux-gnu/libGL.so.1.2.0

Reviewer 2: As far as I understand this is going to break a modern system.
Whatever the case, the system is already broken if this is necessary.
```suggestion
```

cmake ..
```

> If cmake fails with a vcpkg error, then delete `/tmp/vircadia/vcpkg`.  

Reviewer 2: I have no info of the /tmp/ directly ever being used for vcpkg, but if you run into vcpkg issues chances are you are hitting a cache that you shouldn't hit (like currently when you try to build with GLES and without GLES without changing the source in between). That cache is in ~/vircadia-files/vcpkg/
```suggestion
If cmake fails with a vcpkg error, then delete `~/vircadia-files/vcpkg/`.  
```

#### Interface

> Running the Interface:

Reviewer: ```suggestion
Running Interface:
```

./interface/interface
```

Go to localhost in the running Interface to visit your newly launched world.

#### Notes

If your goal is to set up a development environment, it is desirable to set the directory that vcpkg builds into with the `HIFI_VCPKG_BASE` environment variable.
For example, you might set `HIFI_VCPKG_BASE` to `/home/$USER/vcpkg`.

> By default, vcpkg will build in the system `/tmp` directory.

Reviewer 2: ```suggestion
By default, vcpkg will build in the `~/vircadia-files/vcpkg/` directory.
```
https://github.com/vircadia/vircadia/blob/d4a832a0c13df1ab1e87737d82bafbd9d3fbdba2/hifi_vcpkg.py#L63
